cr-usecases/src/main/java/com/thefuntasty/mvvm/crusecases/CoroutineScopeOwner.kt

```diff
@@ -193,6 +230,47 @@ interface CoroutineScopeOwner {
             .launchIn(coroutineScope)
     }
 
+    /**
+     * Launch suspend [block] in [coroutineScope]. Encapsulates this call with try catch block and when an exception is thrown
+     * then it is logged in [logUnhandledException] and handled by [defaultErrorHandler].
+     *
+     * If exception is [CancellationException] then [defaultErrorHandler] is not called and [logUnhandledException] is called
+     * only if the root cause of this exception is not [CancellationException] (e.g. when [Result.getOrCancel] is used).
+     */
+    fun launchWithHandler(block: suspend CoroutineScope.() -> Unit) {
+        coroutineScope.launch {
+            try {
+                block()
+            } catch (exception: Throwable) {
+                if (exception is CancellationException) {
```

Comment:
<table data-meta="generated_by_danger-pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/1e22aaaf9def5bbeda391f2cd9735be2fc018fbd/cr-usecases/src/main/java/com/thefuntasty/mvvm/crusecases/CoroutineScopeOwner.kt#L245"></span>Instead of checking for a general exception type and checking for a specific exception type use multiple catch blocks.</td>
    </tr>
  </tbody>
</table>